types/connect/connect-tests.ts

```diff
@@ -1,45 +1,65 @@
+// npm i --no-save connect
+// tsc --noEmit false && node connect-tests.js
+
 import * as http from "http";
 import connect = require("connect");
 
 const app = connect();
 
 // log all requests
-app.use((req: http.IncomingMessage, res: http.ServerResponse, next: connect.NextFunction) => {
```

Comment:
We're ripping out a lot of tests that ensure that `req` can be an `IncomingMessage`. I still think that should be valid, so could you make sure all of the cases are tested with both types?